src/subapps/projects/hooks/useActivitySubClasses.ts

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { useNexusContext } from '@bbp/react-nexus';
+import { Resource, ResourceLink, PaginatedList } from '@bbp/nexus-sdk';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../../shared/store/reducers';
+
+import fusionConfig from '../config';
+import { isSubClass } from '../utils';
+import { displayError } from '../components/Notifications';
+
+export const useActivitySubClasses = () => {
+  const nexus = useNexusContext();
+  const dataModelsLocation = useSelector(
+    (state: RootState) => state.config.dataModelsLocation
+  );
+  const [subClassesIds, setsubClassesIds] = React.useState<string[]>([]);
+  const [subClasses, setSubClasses] = React.useState<
+    {
+      label: string;
+      ['@id']: string;
+    }[]
+  >([]);
+
+  const [datamodelsOrg, datamodelsProject] = dataModelsLocation.split('/');
+
+  const { datamodelsActivityId } = fusionConfig;
+
+  React.useEffect(() => {
```

Comment:
if you use `useAsyncCall` here instead, and pass the loading and error props to the component that's using this hook, you can seperate the presentation logic from the fetching logic.

Reply:
I haven't use it here as I want to keep values of error and loading in the hook